### ISSUE TITLE
aws_dynamo_db_table throws validation error when disabling stream where stream_view_type is also specified

### ISSUE THREAD

**Issue by kevyau** [terraform-providers/terraform-provider-aws#1134]
### Terraform Version
0.9.8

### Affected Resource(s)
- aws_dynamodb_table

### Terraform Configuration Files
```hcl
resource "aws_dynamodb_table" "table" {
  name = "table"
  read_capacity = 10
  write_capacity = 10
  hash_key = "_id"
  attribute {
    name = "_id"
    type = "N"
  }

  stream_enabled = "${var.stream_enabled}"
  stream_view_type = "${var.stream_view_type}"
}

variable "stream_enabled" {
  default = true
}
variable "stream_view_type" {
  default = "NEW_IMAGE"
}
```

### Error Output
```
* aws_dynamodb_table.table: 1 error(s) occurred:

* aws_dynamodb_table.table: ValidationException: One or more parameter values were invalid: If stream is being disabled, then UpdateViewType must not be specified
```

### Expected Behavior

Assuming previously created dynamodb table with stream enabled
Plan Output:
```
~ aws_dynamodb_table.table
    stream_enabled: "true" => "false"
```
when running apply where `var.stream_enabled` set to `false`, resource should update stream to disabled in AWS

### Actual Behavior
Validation error is thrown due to the presence of `var.stream_view_type` even though the value is the same 

### Steps to Reproduce
1. `terraform apply` to create the dynamodb table and stream in AWS, where variables are defaults 
2. run `terraform apply` where `var.stream_enabled` overridden to `false`

**Comment by rafaljanicki** [comment]
Jsut ran into the issue as this also happens if you remove the `stream_view_type` which is a change apparently not visible to the Terraform

**Comment by radeksimko** [comment]
Hi folks 👋 
the reason you're getting the mentioned `ValidationException` is because we just mirror the API's behaviour here - i.e. we do _not_ ignore the view type when `stream_enabled = false`. 😉 

There are no plans on changing this as we've been bitten many times in the past when trying to build abstractions around API's behaviours. Feel free to raise a support ticket to AWS if you believe the API's behaviour should be changed.

Parametrising stream spec (including disabling it) should however be allowed which is why I raised https://github.com/terraform-providers/terraform-provider-aws/pull/3197 to address that.

After merging ^ you'll be able to disable the stream by setting `stream_enabled = false` **and** `stream_view_type = ""` (empty string).

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.9.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#190-february-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
